cub/cub/thread/thread_reduce.cuh

```diff
@@ -69,23 +127,32 @@ template <int LENGTH,
           typename ReductionOp,
           typename PrefixT,
           typename AccumT = detail::accumulator_t<ReductionOp, PrefixT, T>>
-_CCCL_DEVICE _CCCL_FORCEINLINE AccumT
+_CCCL_DEVICE
+_CCCL_FORCEINLINE ::cuda::std::__enable_if_t<!enable_dpx_reduction<LENGTH, T, ReductionOp, PrefixT, AccumT>(), AccumT>
 ThreadReduce(T* input, ReductionOp reduction_op, PrefixT prefix, Int2Type<LENGTH> /*length*/)
```

Comment:
Suggestion: I always find constraints more readable in the template head:
```c++
template <int LENGTH,
          typename T,
          typename ReductionOp,
          typename PrefixT,
          typename AccumT = detail::accumulator_t<ReductionOp, PrefixT, T>,
          ::cuda::std::__enable_if_t<!enable_dpx_reduction<LENGTH, T, ReductionOp, PrefixT, AccumT>(), int> = 0
>
_CCCL_DEVICE _CCCL_FORCEINLINE
AccumT ThreadReduce(T* input, ReductionOp reduction_op, PrefixT prefix, Int2Type<LENGTH> /*length*/)
```
Applies to other places too.

Reply:
Wow, that's interesting! Do you have a micro-benchmark for that? How did you measure it?

Reply:
well, I didn't perform micro-benchmarking for this case. It mostly based on some blog posts, e.g. https://quuxplusone.github.io/blog/2021/09/04/enable-if-benchmark/

Reply:
also this one is interesting https://github.com/fmtlib/fmt/issues/1023